native/src/main/java/io/ballerina/stdlib/udp/Dispatcher.java

```diff
@@ -75,44 +80,51 @@ public static void invokeOnError(UdpService udpService, String message) {
                     filter(m -> m.getName().equals(Constants.ON_ERROR)).findFirst().orElse(null);
             if (methodType != null) {
                 Object[] params = getOnErrorSignature(message);
-                invokeAsyncCall(udpService.getService(), Constants.ON_ERROR, udpService.getRuntime(),
-                        new UdpCallback(udpService), params);
+                invokeAsyncCall(udpService, null, null, Constants.ON_ERROR, params);
             }
         } catch (Throwable t) {
             log.error("Error while executing onError function", t);
         }
     }
 
-    private static void invokeAsyncCall(BObject service, String methodName, Runtime runtime, UdpCallback callback,
-                                        Object[] params) {
-        StrandMetadata metadata = new StrandMetadata(Utils.getModule().getOrg(), Utils.getModule().getName(),
-                Utils.getModule().getVersion(), methodName);
-        ObjectType objectType = (ObjectType) TypeUtils.getReferredType(TypeUtils.getType(service));
-        if (objectType.isIsolated() && objectType.isIsolated(methodName)) {
-            runtime.invokeMethodAsyncConcurrently(service, methodName,
-                    null, metadata, callback, null, null, params);
-        } else {
-            runtime.invokeMethodAsyncSequentially(service, methodName,
-                    null, metadata, callback, null, null, params);
-        }
+    private static void invokeAsyncCall(UdpService udpService, DatagramPacket datagramPacket, Channel channel,
+                                        String methodName, Object[] params) {
+        Thread.startVirtualThread(() -> {
+            BObject service = udpService.getService();
```

Comment:
Since Dispatcher is call from non ballerina thread (UdpListenerHandler netty thread) We should not block any calls